### PR TITLE
LPS-182148 Create Encrypted ObjectFieldBusinessType

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPS-182148

### CODE REVIEW

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:object

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
Just starting reviewing :)

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/EncryptedObjectFieldBusinessType.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+)
+public class EncryptedObjectFieldBusinessType
+	extends BaseObjectFieldBusinessType {
+
+	@Override
+	public Set<String> getAllowedObjectFieldSettingsNames() {
+		return SetUtil.fromArray(
+			ObjectFieldSettingConstants.NAME_MAX_LENGTH,
+			ObjectFieldSettingConstants.NAME_SHOW_COUNTER);
+	}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/EncryptedObjectFieldBusinessType.java:51]: @p-albuquerque do we need this settings?

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/EncryptedObjectFieldBusinessType.java (hunk 1 above)
Comment (p-albuquerque) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/EncryptedObjectFieldBusinessType.java:51]: According to the mockup of the encrypted field edit sidebar, this field is able to support limit characters option, to this end, I set this settings support.
image

Comment (MarinhoFeliphe) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/EncryptedObjectFieldBusinessType.java:51]: got it, thank you @p-albuquerque

**Comment by naraandrade** [comment]
Just started reviewing :)

**Pull request closed without merging**